awx/api/serializers.py

```diff
@@ -3230,17 +3230,20 @@ class JobRelaunchSerializer(BaseSerializer):
         ],
         write_only=True
     )
+    credential_passwords = VerbatimField(required=True, write_only=True)
```

Comment:
😁

Reply:
btw, now that you comment...

https://github.com/ansible/awx/blob/4c2cff7a63ce0dada14b977b1949cf7d497f1880/awx/api/serializers.py#L556

VerbatimField does not get included in `get_validation_exclusions`, and I'm not sure what the intent is. Was the intent of that clause to check `if isinstance(field, serializers.Field):`? Or is there some weird case where a field is also a serializer that should be excluded?

Reply:
That code was [borrowed from DRF 2.4](https://github.com/encode/django-rest-framework/blob/version-2.4.x/rest_framework/serializers.py#L942) when we transitioned to DRF 3.x , since [DRF 3.x removed the implicit use of model clean methods, which we using in a number of cases](http://www.django-rest-framework.org/topics/3.0-announcement/#differences-between-modelserializer-validation-and-modelform).